src/arg_parsers/gravity.rs

```diff
@@ -0,0 +1,69 @@
+use crate::arg_parse_err::ArgParseErr;
+use std::{ffi::OsStr, str::FromStr};
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Gravity {
+    NorthWest,
+    North,
+    NorthEast,
+    West,
+    Center,
+    East,
+    SouthWest,
+    South,
+    SouthEast,
+}
+
+impl FromStr for Gravity {
+    type Err = ArgParseErr;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::try_from(OsStr::new(s))
+    }
+}
+
+impl TryFrom<&OsStr> for Gravity {
```

Comment:
`strum.EnumString` can take care of this for you -- perhaps not from OsStr, but at least from str. Have a look at how it's used in the `Filter` enum definition.